### DISCUSSION TITLE
signIn doesn't redirect. Promise is stuck at pending

### DISCUSSION THREAD

**Discussion by bardsol** [nextauthjs/next-auth#4756] [Help]
### Question 💬

My signIn function started not redirecting after a successfull signIn, and I don't think I have done anything different.

Even if I set redirect to false the promise returned from signIn never completes. It's stuck at pending.

Anyone have any ideas what may be causing it?

### How to reproduce ☕️

My login form:

```
<Form
        className={className}
        onSubmit={async ({ email }) => {
          const res = await signIn('email', { email, redirect: true});
        }}
      >...
</Form>
```

```
<Form
        className={className}
        onSubmit={async ({ email }) => {
          signIn('email', { email, redirect: true});
        }}
      >...
</Form>
```

```
<Form
        className={className}
        onSubmit={async ({ email }) => {
          const res = await signIn('email', { email, redirect: false});
          console.log(res) // Never logs
        }}
      >...
</Form>
```

[...nextauth].js:
```
import NextAuth from 'next-auth';
import EmailProvider from 'next-auth/providers/email';

import { sendVerificationRequest } from '@modules/next-auth/send-email';
import { prisma } from '@modules/prisma/prisma';
import { PrismaAdapter } from '@next-auth/prisma-adapter';

export default NextAuth({
  adapter: PrismaAdapter(prisma),
  providers: [
    EmailProvider({
      id: 'email',
      server: process.env.EMAIL_SERVER,
      from: process.env.EMAIL_FROM,
      name: process.env.EMAIL_FROM_NAME,
      sendVerificationRequest: sendVerificationRequest,
    }),
    EmailProvider({
      id: process.env.PROVIDER_KEY,
      server: process.env.EMAIL_SERVER,
      from: process.env.EMAIL_FROM,
      name: process.env.EMAIL_FROM_NAME,
      sendVerificationRequest: sendVerificationRequest,
    }),
  ],
  pages: {
    signIn: '/logg-inn',
    signOut: '/logg-ut',
    verifyRequest: '/verifiser-epost',
    newUser: '/registrering',
  },
});
```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by LOGOLFGODORD** [comment]
I team member should be with you shortlySent from Mail for Windows From: Bård SolvangSent: Thursday, June 23, 2022 2:45 AMTo: nextauthjs/next-authCc: SubscribedSubject: [nextauthjs/next-auth] signIn doesn't redirect. Promise is stuck at pending (Issue #4755) Question 💬My signIn function started not redirecting after a successfull signIn, and I don't think I have done anything different.Even if I set redirect to false the promise returned from signIn never completes. It's stuck at pending.Anyone have any ideas what may be causing it?How to reproduce ☕️My login form:<Form className={className} onSubmit={async ({ email }) => { const res = await signIn('email', { email, redirect: false }); console.log(res); // Never logs }} >... </Form[...nextauth].js:`import NextAuth from 'next-auth';import EmailProvider from 'next-auth/providers/email';import { sendVerificationRequest } from ***@***.***/next-auth/send-email';import { prisma } from ***@***.***/prisma/prisma';import { PrismaAdapter } from ***@***.***/prisma-adapter';export default NextAuth({adapter: PrismaAdapter(prisma),providers: [EmailProvider({id: 'email',server: process.env.EMAIL_SERVER,from: process.env.EMAIL_FROM,name: process.env.EMAIL_FROM_NAME,sendVerificationRequest: sendVerificationRequest,}),EmailProvider({id: process.env.PROVIDER_KEY,server: process.env.EMAIL_SERVER,from: process.env.EMAIL_FROM,name: process.env.EMAIL_FROM_NAME,sendVerificationRequest: sendVerificationRequest,}),],pages: {signIn: '/logg-inn',signOut: '/logg-ut',verifyRequest: '/verifiser-epost',newUser: '/registrering',},});`Contributing 🙌🏽No, I am afraid I cannot help regarding this—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>
